src/diem/miniwallet/app/app.py

```diff
@@ -0,0 +1,263 @@
+            self._execute_send_payment_txns()
+
+    def get_resources(self, typ: Type[T]) -> List[T]:
+        return self.store.find_all(typ)
+
+    def create_resource(self, typ: Type[T], data: JsonInput) -> T:
+        with self.store.transaction:
+            return getattr(self, "_create_resource_%s" % typ.resource_name())(data)
+
+    def _create_resource_account(self, data: JsonInput) -> Account:
+        return self.store.create(
+            Account,
+            kyc_data=data.get("kyc_data", str, self._validate_kyc_data),
+        )
+
+    def _create_resource_transaction(self, data: JsonInput) -> Transaction:
+        account = self.store.find(Account, id=data.get("account_id", str))
+        payee = data.get_nullable("payee", str, self._validate_account_identifier)
+        return self.store.create(
+            Transaction,
+            account_id=account.id,
+            currency=data.get("currency", str, self._validate_currency_code),
+            amount=data.get("amount", int, self._validate_amount),
+            payee=payee,
+            status="pending" if payee else "completed",
+            before_create=self._validate_account_balance,
+        )
+
+    def _create_resource_payment_uri(self, data: JsonInput) -> PaymentURI:
+        subaddress = self._gen_subaddress()
+        return self.store.create(
+            PaymentURI,
+            account_id=self.store.find(Account, id=data.get("account_id", str)).id,
+            subaddress_hex=subaddress.hex(),
+            account_identifier=self.diem.account.account_identifier(subaddress),
+        )
+
+    def _execute_send_payment_txns(self) -> None:
+        for txn in self.store.find_all(Transaction, status="pending"):
+            if txn.payee is not None:
+                self._execute_send_payment_txn(txn)
+
+    def _execute_send_payment_txn(self, txn: Transaction) -> None:
+        if self.offchain.is_my_account_id(str(txn.payee)):
+            self._send_internal_payment_txn(txn)
+        else:
+            self._send_external_payment_txn(txn)
+
+    def _send_internal_payment_txn(self, txn: Transaction) -> None:
+        _, payee_subaddress = self.diem.account.decode_account_identifier(str(txn.payee))
+        rp = self.store.find(PaymentURI, subaddress_hex=utils.hex(payee_subaddress))
+        self.store.create(
+            Transaction, account_id=rp.account_id, currency=txn.currency, amount=txn.amount, status="completed"
+        )
+        txn.status = "completed"
+
+    def _send_external_payment_txn(self, txn: Transaction) -> None:
+        if txn.signed_transaction:
+            try:
+                diem_txn = self.diem.client.wait_for_transaction(str(txn.signed_transaction))
+                txn.status = "completed"
+                txn.diem_transaction_version = diem_txn.version
+            except jsonrpc.WaitForTransactionTimeout:
+                pass  # need continue to wait
+            except (jsonrpc.TransactionHashMismatchError):
```

Comment:
remove `()`
also when could this happen?

Reply:
this happens when you submit transaction with same sequence_number set for the transaction.
The reason you may get same sequence number is because you can only execute a transaction that sequence_number is equal to diem account sequence number. 
you can get account sequence number by calling get_account json rpc api.
it is hard to manage when you create transaction, and you don't know how many transactions are on fly.